### PR TITLE
Refactor gp model microservice

### PR DESCRIPTION
**Pull request by lilly-muyama**
# <Refactoring GP Model>

**_WHAT DOES THIS PR DO?_**
This PR creates a microservice for the GP model cloud function

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[DP-44](https://airqoteam.atlassian.net/jira/software/projects/DP/boards/44?assignee=5d91eed2d8794c0d88270106&selectedIssue=DP-44)
**_WHAT IS THE LINK TO THE PR BRANCH_**

**_HOW DO I TEST OUT THIS PR?_**
- Git clone this repo
- cd src/gp-model
- env\scripts\activate for Windows or source env/bin/activate for MacOS
- pip install -r requirements.txt
- python cloud_function.py

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
N/A

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

### CODE REVIEW

**Comment by lilly-muyama** [comment]
@danielogen could you please help create the docker files for this micro-service since there are always issues with installation of the python requirements? Thank you.

**Comment by danielogen** [comment]
> @danielogen could you please help create the docker files for this micro-service since there are always issues with installation of the python requirements? Thank you.

Yes @lilly-muyama , We might have to take sometime to review the requirement file and remove any package which was carried along from the predict microservice

**Comment by lilly-muyama** [comment]
> 
> 
> > @danielogen could you please help create the docker files for this micro-service since there are always issues with installation of the python requirements? Thank you.
> 
> Yes @lilly-muyama , We might have to take sometime to review the requirement file and remove any package which was carried along from the predict microservice

@danielogen Okay, let me work on that

**Comment by lilly-muyama** [comment]
@danielogen I have updated the requirements.txt file

**Comment by danielogen** [comment]
> @danielogen I have updated the requirements.txt file

Thank you @lilly-muyama.  I'm having a very bad internet day but I will work on this tonight.

File: src/gp-model/cloud_function.py [hunk 1]
```diff
@@ -15,13 +15,16 @@
...
 from dotenv import load_dotenv
-load_dotenv()
+import math
+import fiona
+import geopandas
 
+load_dotenv()
 
-storage_client = storage.Client('AirQo-e37846081569.json')
 MONGO_URI = os.getenv("MONGO_URI")
```
Comment (danielogen) [inline src/gp-model/cloud_function.py:24]: just curious about how we can easily handle multiple environments here

**Review by Baalmart — COMMENTED** [review]
Thanks @lilly-muyama and @danielogen 
Getting the attached error on package installation. Tried to clear the pip cache but no success, error (`Cache entry deserialization failed, entry ignored`) remains.

Screenshot 2021-07-27 at 17 09 11

**Comment by lilly-muyama** [comment]
@Baalmart The `Cache entry deserialization failed, entry ignored` is just a warning and not an error. Did you try `rm -rf ~/.cache/pip`. Also, apparently, it can be a network issue so maybe try later.

However, the error that stops the code running is `Cannot cythonize without Cython installed` which I've never encountered in my life. It must be a Linux/MAC thing so over to @danielogen

**Comment by danielogen** [comment]
> @Baalmart The `Cache entry deserialization failed, entry ignored` is just a warning and not an error. Did you try `rm -rf ~/.cache/pip`. Also, apparently, it can be a network issue so maybe try later.
> 
> However, the error that stops the code running is `Cannot cythonize without Cython installed` which I've never encountered in my life. It must be a Linux/MAC thing so over to @danielogen

Could the related to `virtual env`. I'm using python `3.7.3`, I always find upgrading `pip` very helpful before installing packages. So, you could try this command `pip install --upgrade pip` within your `virtualenv`

**Review by Baalmart — APPROVED** [review]
PR works well.
Just left a few comments/suggestions some time back.
I hope we can transfer them into Github issues after/before deployment works 
cc: @danielogen @lilly-muyama

**Review by danielogen — APPROVED** [review]
LGTM

**Pull request merged**